packages/builder/src/components/automation/SetupPanel/RowSelectorTypes.svelte

```diff
@@ -35,153 +36,210 @@
     FieldType.SIGNATURE_SINGLE,
   ]
 
-  let previousBindingState = useAttachmentBinding
-
   function schemaHasOptions(schema) {
     return !!schema.constraints?.inclusion?.length
   }
 
   function handleAttachmentParams(keyValueObj) {
     let params = {}
+    if (!keyValueObj) {
+      return null
+    }
 
     if (!Array.isArray(keyValueObj) && keyValueObj) {
       keyValueObj = [keyValueObj]
     }
 
     if (keyValueObj.length) {
       for (let param of keyValueObj) {
-        params[param.url] = param.filename
+        params[param.url || ""] = param.filename || ""
```

Comment:
Might be missing something, but will this not potentially just add a random empty string key to the params object? Does a guard or something make more sense? 

```js
for (let param of keyValueObj) {
     if (!params.url) {
       continue;
     }
     params[param.url] = param.filename || ""
}
```

Reply:
The main goal here was to just not get `undefined` in the url text box when adding an attachment for the first time. The empty string is just a starting point for the url. The main reason why I don't like the above approach is that if you delete the url, it completely removes the entry from the attachments list, even if you've entered your file name.